Jenkinsfile

```diff
@@ -1,28 +1,69 @@
 pipeline {
     agent any
 
+    tools { 
+        maven 'maven_3_5_4' 
+        jdk 'jdk8' 
+    }
+    environment {
+        // requires the Pipeline Utility Steps plugin
+        pom = readMavenPom file: 'pom.xml'
+        pomVersion = pom.getVersion()
+        pomArtifactId = pom.getArtifactId()
+        //pomReleases = pom.getProperties().get("artifactory.releases")
+        //pomSnapshots = pom.getProperties().get("artifactory.snapshots")
+    }
+
     stages {
-        stage('Package') {
+        stage('Rest Repo') {
+            // it is not clear that the repo is a clean checkout
             steps {
-                withMaven(maven:'maven_3_5_4') {
-                    sh 'mvn clean package'
-                }
+                // remove potentially old release files if exist
+                sh 'rm pom.xml.releaseBackup release.properties 2>/dev/null || true'
+                // rest the git state
+                sh 'git checkout -f master'
+                sh 'git reset --hard'
             }
         }
-        stage('Test') {
+        stage('Build/Test') {
             steps {
-                withMaven(maven:'maven_3_5_4') {
-                    sh 'mvn test'
+                sh 'mvn clean package test'
+            }
+            post {
+                success {
+                    junit 'target/surefire-reports/**/*.xml' 
                 }
             }
         }
-        stage('Maven Install') {
+        stage('Release') {
+            // only release build when triggered
+            when {
+                expression{ params.RELEASE_BUILD }
+            }
             steps {
-                withMaven(maven:'maven_3_5_4') {
-                    sh 'mvn install'
+                // incorporate the maven dryRun flag
+                script {
+                    if (params.DRY_RUN) {
+                        dryRun='-DdryRun=true'
+                    } else {
+                        dryRun=''
+                    }
+                    releaseVersion = pomVersion.replace("-SNAPSHOT","")
```

Comment:
Alas... I feel like this shouldn't be necessary, but it obviously is. We're using the Python's `bumpversion` package to increment the version and handle removing `-SNAPSHOT` on NLDI, but that's somewhat different use case because we don't really care about saving the .jar as much.